solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -115,11 +116,46 @@ pub mod solana_ibc {
         ctx.accounts.chain.set_stake((*ctx.accounts.sender.key).into(), amount)
     }
 
+    /// Called to set up escrow and mint accounts for given channel and denom.
+    #[allow(unused_variables)]
+    pub fn deliver_init_escrow<'a, 'info>(
+        ctx: Context<'a, 'a, 'a, 'info, DeliverInitEscrow<'info>>,
+        port_id: ibc::PortId,
+        channel_id_on_b: ibc::ChannelId,
+        base_denom: String,
+    ) -> Result<()> {
+        Ok(())
+    }
+
+    #[allow(unused_variables)]
     pub fn deliver<'a, 'info>(
         ctx: Context<'a, 'a, 'a, 'info, Deliver<'info>>,
+        port_id: Option<ibc::PortId>,
+        channel_id_on_b: Option<ibc::ChannelId>,
+        base_denom: Option<String>,
```

Comment:
So the reason for those arguments is that we need them for seeds? And their value, if set, is the same as one specified somewhere inside of message?

Reply:
So we either check validate the seeds here or we do in send validate. If we do in send validate, there would not be any need of these params which would be better IMO.

Reply:
We need those values in `seeds` annotation though, no?

Reply:
It's not necessary, we can skip it. It's only for validation which we can do it in send validate.

Reply:
SGTM